core/estring.cpp

```diff
@@ -946,7 +946,15 @@ uint EString::number( bool * ok, uint base ) const
 EString EString::fromNumber( int64 n, uint base )
 {
     EString r;
-    r.appendNumber( n, base );
+    if (0 == n) {
+        r.append("0");        // Short-circuit, 0 is 0 in any base, no need for extra processing
```

Comment:
I bet this is actually a net loss. The cost of the extra if() in all cases is more than the savings on the 0.